src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1221,486 +1643,612 @@ s32 func_80A13B1C(EnGo* this, PlayState* play) {
+    this->actionFlags = ENGO_FLAG_NONE;
+    this->actor.gravity = ENGO_NORMAL_GRAVITY;
+    SubS_UpdateFlags(&this->actionFlags, 3, 7);
+    this->sleepState = ENGO_AWAKE;
+    this->actionFlags |= ENGO_FLAG_LOST_ATTENTION;
+    this->eyeTimer = 0;
+    this->eyeTexIndex = ENGO_EYETEX_CLOSED2;
+    this->iceBlockScale = 0.0f;
+    this->iceBlockAlpha = 0.0f;
+}
+
+/**
+ * Update the Goron's Animation and Action Flags to Shivering.
+ *
+ * @see EnGo_Idle
+ */
+void EnGo_UpdateToShivering(EnGo* this, PlayState* play) {
+    EnGo_ChangeAnim(this, play, ENGO_ANIM_SHIVER);
+    Actor_SetScale(&this->actor, this->scaleFactor);
+    this->actionFlags = ENGO_FLAG_NONE;
+    this->actor.gravity = ENGO_NORMAL_GRAVITY;
+    SubS_UpdateFlags(&this->actionFlags, 3, 7);
+    this->sleepState = ENGO_AWAKE;
+    this->actionFlags |= ENGO_FLAG_LOST_ATTENTION;
+    this->actionFlags |= ENGO_FLAG_EYESOPEN;
+    this->eyeTimer = 0;
+    this->eyeTexIndex = ENGO_EYETEX_OPEN;
+    this->iceBlockScale = 0.0f;
+    this->iceBlockAlpha = 0.0f;
+}
+
+/**
+ * Setup a goron stretching at the racetrack.
+ *
+ * Stretching gorons placed in the starting area of the racetrack doing various stretches.
+ */
+void EnGo_StretcherGoron_Setup(EnGo* this, PlayState* play) {
     if (((gSaveContext.save.entrance == ENTRANCE(GORON_RACETRACK, 0)) ||
          (gSaveContext.save.entrance == ENTRANCE(GORON_RACETRACK, 2))) &&
         (gSaveContext.save.weekEventReg[33] & 0x80)) {
-        func_80A14018(this, play);
-        this->actionFunc = func_80A149B0;
+        EnGo_UpdateToStretching(this, play);
+        this->actionFunc = EnGo_Idle;
     } else {
         Actor_MarkForDeath(&this->actor);
     }
 }
 
-void func_80A1449C(EnGo* this, PlayState* play) {
+/**
+ * Setup a goron spectating at the racetrack.
+ *
+ * Spectators to the goron races cannot be engaged, they simply stand idle and cheer.
+ */
+void EnGo_SpectatorGoron_Setup(EnGo* this, PlayState* play) {
     if ((gSaveContext.save.entrance == ENTRANCE(GORON_RACETRACK, 1)) ||
         (gSaveContext.save.entrance == ENTRANCE(CUTSCENE, 0))) {
-        func_80A14104(this, play);
-        this->actionFunc = func_80A149B0;
+        EnGo_UpdateToSpectating(this, play);
+        this->actionFunc = EnGo_Idle;
     } else {
         Actor_MarkForDeath(&this->actor);
     }
 }
 
-void func_80A144F4(EnGo* this, PlayState* play) {
+/**
+ * Setup the Goron Shrine GateKeeper.
+ *
+ * When spoken to, can open the Goron Shrine for the player.
+ * - On Day 1: Can be found in the open, shivering
+ * - On Days 2,3: Can be found in a snowball.
+ */
+void EnGo_GatekeeperGoron_Setup(EnGo* this, PlayState* play) {
     if (gSaveContext.save.day >= 2) {
-        this->unk_284 = SubS_GetDayDependentPath(play, ENGO_GET_7F80(&this->actor), 0xFF, &this->unk_3E4);
-        if (this->unk_284 != NULL) {
-            this->unk_3E4 = 1;
+        this->gatekeeperPath = SubS_GetDayDependentPath(play, ENGO_GET_7F80(&this->actor), 0xFF, &this->indexPathPoint);
```

Comment:
Based on usage here, should be able to name `ENGO_GET_7F80`. Something like `ENGO_GET_PATH`